wurzel/backend/backend_argo.py

```diff
@@ -0,0 +1,274 @@
+
+    Attributes:
+        bucket (str): Name of the S3 bucket used to store pipeline artifacts.
+        endpoint (str): Endpoint URL of the S3-compatible storage service.
+
+    """
+
+    bucket: str = "wurzel-bucket"
+    endpoint: str = "s3.amazonaws.com"
+
+
+DNS_LABEL_REGEX = r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
+
+
+class ArgoBackendSettings(SettingsBase):
+    """Settings object for the Argo Workflows backend, configurable via environment variables.
+
+    Environment Variables:
+        Prefix: ARGOWORKFLOWBACKEND__
+
+    Attributes:
+        IMAGE (str): Docker image used to run pipeline steps.
+        INLINE_STEP_SETTINGS (bool): if STEP environment variables have to get loaded based on settings and passed directly to it's step.
+        SCHEDULE (str): Cron expression for scheduling the workflow.
+        DATA_DIR (Path): Base directory inside the container where step data is written.
+        ENCAPSULATE_ENV (bool): Whether to encapsulate step execution with environment variables.
+        S3_ARTIFACT_TEMPLATE (S3ArtifactTemplate): S3 configuration used for input/output mapping.
+        SERVICE_ACCOUNT_NAME (str): Kubernetes service account used for Argo workflow execution.
+        SECRET_NAME (str): Name of the Kubernetes Secret passed into the workflow container.
+        CONFIG_MAP (str): Name of the ConfigMap passed into the container environment.
+        ANNOTATIONS (dict): Custom annotations to be applied to workflow tasks.
+        NAMESPACE (str): Kubernetes namespace to run the workflow in.
+        PIPELINE_NAME (str): Name of the Argo workflow pipeline (must comply with DNS label rules).
+
+    """
+
+    model_config = SettingsConfigDict(env_prefix="ARGOWORKFLOWBACKEND__")
+    INLINE_STEP_SETTINGS: bool = False
+    IMAGE: str = "ghcr.io/telekom/wurzel"
+    SCHEDULE: str = "0 4 * * *"
+    DATA_DIR: Path = Path("/usr/app")
+    ENCAPSULATE_ENV: bool = True
+    S3_ARTIFACT_TEMPLATE: S3ArtifactTemplate = S3ArtifactTemplate()
+    SERVICE_ACCOUNT_NAME: str = "wurzel-service-account"
+    SECRET_NAME: str = "wurzel-secret"
+    CONFIG_MAP: str = "wurzel-config"
+    ANNOTATIONS: dict = {"sidecar.istio.io/inject": "false"}
+    NAMESPACE: str = "argo-workflows"
+    PIPELINE_NAME: str = Field(
+        default="wurzel",
+        max_length=63,
+        description="Kubernetes-compliant name: lowercase alphanumeric, '-' allowed, must start and end with alphanumeric.",
+    )
+
+    @field_validator("PIPELINE_NAME")
+    @classmethod
+    def _validate_pipeline_name(cls, value: str) -> str:
+        pattern = r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
+        if not re.fullmatch(pattern, value):
+            raise ValueError(
+                "PIPELINE_NAME must consist of lower-case letters, numbers, and '-', "
+                "start and end with a letter or number, and be <= 63 characters."
+            )
+        return value
```

Comment:
```suggestion
    PIPELINE_NAME: str = Field(
        default="wurzel",
        max_length=63,
        description="Kubernetes-compliant name: lowercase alphanumeric, '-' allowed, must start and end with alphanumeric.",
        pattern=r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
    )
```
Should work the same, less code